triggers/service/src/main/scala/com/digitalasset/daml/lf/engine/trigger/AuthServiceClient.scala

```diff
@@ -0,0 +1,70 @@
+package com.daml.lf.engine.trigger
+
+import java.net.InetAddress
+
+import akka.actor.ActorSystem
+import akka.http.scaladsl.{Http, HttpExt}
+import akka.http.scaladsl.model.{HttpMethods, HttpRequest, HttpResponse}
+import akka.http.scaladsl.model.headers.{Authorization, BasicHttpCredentials}
+import akka.stream.Materializer
+import akka.util.ByteString
+import com.daml.ports.Port
+import spray.json.{JsObject, JsValue}
+import spray.json._
+
+import scala.concurrent.{ExecutionContext, Future}
+
+// OAuth2 bearer token for requests to authentication service
+case class AuthServiceToken(token: String)
+
+class AuthServiceClient(
+    host: InetAddress,
+    port: Port,
+)(implicit system: ActorSystem, materializer: Materializer, ec: ExecutionContext) {
+
+  private val http: HttpExt = Http(system)
+
+  private val authServiceBaseUri = "http://" + host.getHostAddress + ":" + port.toString + "/sa/"
+
+  private def responseBodyToString(resp: HttpResponse): Future[String] = {
+    resp.entity.dataBytes.runFold(ByteString(""))(_ ++ _).map(_.utf8String)
+  }
+
+  // Extract a field from the JSON object in an HTTP response.
+  private def parseResponseField(response: HttpResponse, field: String): Future[Option[JsValue]] = {
```

Comment:
Is there a reason why you are not defining a proper `RootJsonReader`?

Reply:
No - will do